### ISSUE TITLE
Using an IRect as clip parameter in Page.get_pixmap no longer works since 1.23.9

### ISSUE THREAD

**Issue by stijnvermeeren-swisstopo** [pymupdf/PyMuPDF#3134]
### Description of the bug

In PyMuPDF version 1.23.9 and newer, an `IRect` is no longer taken into account when provided as the `clip` parameter for the `Page.get_pixmap` method. Providing a `Rect` still works.

### How to reproduce the bug

```
import fitz

doc = fitz.Document()
page = doc.new_page()
page.get_pixmap(clip=fitz.Rect(0, 0, 100, 100)).save("test_rect.jpg")
page.get_pixmap(clip=fitz.IRect(0, 0, 100, 100)).save("test_irect.jpg")
```

**Expected behavior / behavior with PyMuPDF 1.23.8**
Two JPG images of size 100x100 are created.

**Behavior with PyMuPDF 1.23.9 and 1.23.21**
The image test_rect.jpg has size 100x100 but the image test_irect.jpg has size 595x842 (full page).

### PyMuPDF version

1.23.21

### Operating system

Linux

### Python version

3.11

**Comment by julian-smith-artifex-com** [comment]
I have a fix in my tree.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.23.
